xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-default/src/main/java/org/xwiki/refactoring/internal/DefaultModelBridge.java

```diff
@@ -440,6 +440,10 @@ public boolean restoreDeletedDocument(long deletedDocumentId, boolean checkConte
                 logger.error("Document [{}] with ID [{}] can not be restored. Document already exists",
                     deletedDocument.getFullName(), deletedDocumentId);
             } else if (checkContextUser && !canRestoreDeletedDocument(deletedDocumentId, context.getUserReference())) {
+                logger.error("The author of this script not allowed to restore document [{}] with ID [{}]",
```

Comment:
I think it would be even better to be even more explicit and indicate the user reference in the message (like `The author ({}) if this...`).

Reply:
Done.